src/script/standard.cpp

```diff
@@ -65,6 +68,12 @@ bool Solver(const CScript& scriptPubKey, txnouttype& typeRet, std::vector<std::v
         // Sender provides N pubkeys, receivers provides M signatures
         mTemplates.insert(std::make_pair(TX_MULTISIG, CScript() << OP_SMALLINTEGER << OP_PUBKEYS << OP_SMALLINTEGER << OP_CHECKMULTISIG));
 
+        // Contract creation tx with sender
+        mTemplates.insert(std::make_pair(TX_CREATE_SENDER, CScript() << OP_ADDRESS_TYPE << OP_ADDRESS << OP_SCRIPT_SIG << OP_SENDER << OP_VERSION << OP_GAS_LIMIT << OP_GAS_PRICE << OP_DATA << OP_CREATE));
```

Comment:
Are there some consensus rules in place to ensure that non-standard transactions with OP_SENDER is invalid and can't be put into a block?

Reply:
The sender data is extracted by `ExtractSenderData` which solves (call function `Solver`) the sender script that match the 2 allowed sender templates. The function is called in several cases.